### DISCUSSION TITLE
LittleFS WAV input, I2S output, controlled by Audio Player

### DISCUSSION THREAD

**Discussion by pauldupuyjr** [pschatzmann/arduino-audio-tools#454] [Q&A] [answered]
Looking for an example of controlling a LittleFS flash stored WAV file with Audio Player, outputting to I2S.

I see a similar issue (SPIFFS support on ESP32? #133), but it isn't clear to me how to explicitly implement the chain and Audio Player.

Thanks in advance for any help.

**Comment by pschatzmann** [comment]
I don't think that this makes a lot of sense: There is hardly enough space for one wav file on an ESP32, therefore I do not provide this out of the box.

I am not a big fan of LittleFS and SPIFFS and I usually prefer to upload "files" as binary data, e.g. see [this example](https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-stream/streams-memory_mp3_short-i2s).

However you can easily implement your own AudioSource subclass (if you want to use the Audio Player). I suggest that you have a look at the existing file based implementations: e.g. AudioSourceSD.

A little FS File is a subclass of Stream, so if you just want to output a single file you can just use it as copy source.

**Comment by pauldupuyjr** [comment]
Thanks for the quick reply Phil. I'm using a ProS3 with 16M of flash; the WAV files are relatively short, in the neighborhood of 100K. And I need to persist and manage multiple files, so the binary data approach won't work for my case. 

Thank you for the pointer to the AudioSource class and SD example.

**Comment by pschatzmann** [comment, marked as answer]
I have just added support for the LittleFS to the AudioPlayer. An example can be found [here](https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-player/player-littlefs-i2s).

**Reply by pauldupuyjr** [reply]
Excellent -- thanks @pschatzmann !
